src/lib.rs

```diff
@@ -40,15 +40,22 @@
 
 extern crate libc;
 
-use libc::{pid_t, c_char, c_void, getpid, size_t, c_int};
+use libc::{pid_t, c_char, c_void, getpid, size_t};
 use std::ffi::CString;
 use std::path::PathBuf;
 
+#[allow(dead_code)]
+#[repr(u8)]
+enum CBool {
+    CFalse = 0,
+    CTrue = 1,
+}
+
 // FFI stubs
 #[link(name = "traceme")]
 extern "C" {
     fn traceme_start_tracer(conf: *const TracerConf) -> *const c_void;
-    fn traceme_stop_tracer(tr_ctx: *const c_void) -> c_int;
+    fn traceme_stop_tracer(tr_ctx: *const c_void) -> u8;
```

Comment:
I guess this is `u8` because the Rust FFI doesn't support bools? If so, are we sure that bools will only ever be 0 or 1? If so, I think this is fine.

Reply:
The Rust FFI doesn't support C bools because Rust doesn't know how big a `bool` is on any given platform. I'm hoping that if a `bool` were really (e.g.) a 4-byte `int` then our use of `u8` would cause it to be truncated to the least significant byte...

Does that make sense?

Reply:
On the other hand, on the post linked above people are saying a C `bool`, could be 1-bit, so we'd be hoping that the Rust FFI does the necessary zero extension :\

I mean, the code works, but I'm not certain it's correct.

Reply:
Why don't we add a check (into a `build.rs`?) that compiles a couple of test cases for this in C so that we have some confidence that it'll keep working into the future (and, if it stops working, we'll be alerted)?

Reply:
OK, `build.rs` would be a bad idea, as they'd be checked all the time. I wonder if there's a way to do a sort of `configure.rs` equivalent?

Reply:
You can do whatever you want in `build.rs` and then set `!cfg` macros, so you can somewhat emulate a configure script. I'd rather avoid anything like that if we can though.

FWIW, `sizeof(bool)` is 1 on my desktop and bencher8.

Here are my two concrete suggestions at this point:
 * Roll with the diff we have, because it will explode loudly at some point if we go it wrong.
 * Or, just use an `int` until rustc gets a well-defined `libc::c_bool`

I'll take either :)

Reply:
My point is more that `build.rs` is invoked regularly, so I'm not sure we would want to do invoke two C compiles every time we do `cargo check`. Although maybe we can cache results somewhere?

Unfortunately, I do think we should add a check that our assumptions are right :(